Mathlib/Algebra/Group/Pointwise/Finset/Basic.lean

```diff
@@ -1588,6 +1599,17 @@ theorem inter_mul_singleton : s ∩ t * {a} = s * {a} ∩ (t * {a}) :=
 theorem card_le_card_mul_right {t : Finset α} (ht : t.Nonempty) : s.card ≤ (s * t).card :=
   card_le_card_image₂_right _ ht mul_left_injective
 
+/--
+The size of `s * s` is at least the size of `s`, version with right-cancellative multiplication.
+See `card_le_card_mul_self` for the version with left-cancellative multiplication.
+-/
+@[to_additive
+"The size of `s + s` is at least the size of `s`, version with right-cancellative addition.
+See `card_le_card_add_self'` for the version with left-cancellative addition."
+]
+theorem card_le_card_mul_self' {s : Finset α} : s.card ≤ (s * s).card := by
```

Comment:
Given the discussion at https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/theorem.20names.20with.20primes.3F, have you considered naming these something like `card_le_card_mul_self_(left|right)`?

Reply:
Indeed that discussion is why I gave these docstrings. I think your suggested name is worse, since the "main" part of the statement here is unchanged in both versions, so the `left`/`right` distinction is more confusing than helpful. Additionally, the most common use-case for these definitions is in a group, where both conditions clearly are satisfied and the distinction is meaningless, and as such `card_le_card_add_self` is a more helpful name than `card_le_card_add_self'`.